### PR TITLE
Use z3 with 10 bit for sol verify

### PR DESCRIPTION
**Pull request by XiangyuG**

### CODE REVIEW

File: chipc/z3_utils.py [hunk 1]
```diff
@@ -105,9 +105,8 @@ def check_without_bnds(smt2_filename):
     # The original formula's body is comprised of Implies(A, B) where A
     # specifies range of input variables and where B is a condition that we're
```
Comment (taegyunkim) [inline chipc/z3_utils.py]: Update comments and the function name as we're simply just checking and not removing the bounds

**Review by taegyunkim — CHANGES_REQUESTED** [review]
Also update the name of test in https://github.com/chipmunk-project/chipmunk/blob/master/tests/test_z3_utils.py

We can also remove strip_input_bounds() function, and its tests.

File: chipc/z3_utils.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def generate_counter_examples(smt2_filename):
     return (pkt_fields, state_vars)
 
 
-def check_without_bnds(smt2_filename):
+def check_with_z3(smt2_filename):
```
Comment (taegyunkim) [inline chipc/z3_utils.py]: The file name is z3_utils it's clear that we are using z3. How about just check?

**Review by taegyunkim — CHANGES_REQUESTED** [review]

File: chipc/z3_utils.py (hunk 1 above)
Comment (XiangyuG) [reply chipc/z3_utils.py]: Done!

File: chipc/z3_utils.py (hunk 2 above)
Comment (XiangyuG) [reply chipc/z3_utils.py]: Yeah, it makes sense. But I think check is somehow too obscure, let me think about it and we can merge it tomorrow.

File: chipc/z3_utils.py [hunk 3]
```diff
@@ -74,40 +74,20 @@ def generate_counter_examples(smt2_filename):
...
-        body), ("Given formula doesn't have z3.Implies as body:\n", body)
-    new_body = formula.body().children()[1]
-    return z3.ForAll(variables, new_body)
-
-
-def check_without_bnds(smt2_filename):
+def simple_check(smt2_filename):
     """Given a smt2 file generated from a sketch, parses assertion from the
-    file and checks whether it holds for all integers representable in z3.
+    file and checks whether it holds for specific range representable in z3.
```
Comment (taegyunkim) [inline chipc/z3_utils.py]:
```suggestion
    file and checks with z3. We assume that the file already has input bit
    ranges defined by sketch. 
```

File: tests/test_z3_utils.py [hunk 4]
```diff
@@ -72,25 +72,9 @@ def test_unsat_formula(self):
...
-    def test_success(self):
-        a = z3.Int('a')
-        input_formula = z3.ForAll([a], z3.Implies(a > 0, a == 0))
-        expected_formula = z3.ForAll([a], a == 0)
-        self.assertEqual(expected_formula,
-                         z3_utils.strip_input_bounds(input_formula))
-
-
-class CheckWithoutBndsTest(unittest.TestCase):
+class CheckWithZ3Test(unittest.TestCase):
```
Comment (taegyunkim) [inline tests/test_z3_utils.py]: It should be SimpleCheckTest, as you've updated the function name to simple_check

**Review by taegyunkim — CHANGES_REQUESTED** [review]

File: tests/test_z3_utils.py (hunk 4 above)
Comment (XiangyuG) [reply tests/test_z3_utils.py]: Oh, I need to change this

**Comment by taegyunkim** [comment]
> Hello @XiangyuG! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:
> 
> * In the file [`chipc/z3_utils.py`](https://github.com/chipmunk-project/chipmunk/blob/19aebd51066c0b65a9c82fd3f47bd8e43b00e3f9/chipc/z3_utils.py):
> 
> > [Line 80:30](https://github.com/chipmunk-project/chipmunk/blob/19aebd51066c0b65a9c82fd3f47bd8e43b00e3f9/chipc/z3_utils.py#L80): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

Remove trailing whitespace in line 80

**Review by taegyunkim — APPROVED** [review]

**Pull request merged**